tests/test_curves.py

```diff
@@ -843,3 +843,66 @@ def test_compute_ndarray_error(self):
         c = SplineXYZCurve(X=R * np.cos(phi), Y=R * np.sin(phi), Z=np.zeros_like(phi))
         with pytest.raises(TypeError):
             c.compute("length", grid=np.linspace(0, 1, 10))
+
+    @pytest.mark.unit
+    def test_discontinuous_splines(self):
+        """Test splines that have break points."""
+        break_indices = [0, 10]
+
+        def test(method, data_key, compare_breaks=True):
+            R = 2
+            phi = 2 * np.pi * np.linspace(0, 1, 1001, endpoint=True) ** 2
+            discontinuous = SplineXYZCurve(
+                X=R * np.cos(phi),
+                Y=R * np.sin(phi),
+                Z=np.zeros_like(phi),
+                knots="arclength",
+                break_indices=break_indices,
+                method=method,
+            )
+
+            continuous = SplineXYZCurve(
+                X=R * np.cos(phi),
+                Y=R * np.sin(phi),
+                Z=np.zeros_like(phi),
+                knots="arclength",
+                break_indices=None,
+                method=method,
+            )
+
+            assert discontinuous.method == method
+            assert continuous.method == method
+
+            grid = LinearGrid(N=100)
+
+            discont_data = discontinuous.compute([data_key], grid=grid)
+            discont_quantity = discont_data[data_key]
+            cont_quantity = continuous.compute(data_key, grid=grid)[data_key]
+
+            if compare_breaks:
+                np.testing.assert_allclose(discont_quantity, cont_quantity, rtol=1e-3)
+            else:
+                # coil quantities aligned with s array so need to find breaks in s
+                is_break_point = np.isin(
+                    discont_data["s"], discontinuous.knots[break_indices]
+                )
+                # don't include break points
+                np.testing.assert_allclose(
+                    discont_quantity[~is_break_point],
+                    cont_quantity[~is_break_point],
+                    rtol=1e-3,
+                )
+                # we know there is 1 break point in s (0th index)
+                assert len(discont_quantity[~is_break_point]) != len(discont_quantity)
+
+        test("linear", "length")
+        # TODO: not sure why this one doesn't pass, first index is nan for discont
+        test("linear", "curvature", compare_breaks=False)
+        # torsion = 0 for breakpoints, but otherwise torsion = NaN, so can't compare
+        test("linear", "torsion", compare_breaks=False)
+
+        test("cubic", "length")
+        # TODO: test doesn't pass for break_indices = [0, 500]
```

Comment:
this test with "cubic" and "curvature" doesn't pass with `break_indices=[0, 500]` because values are not within the tolerance but does pass with `break_indices=[0, 10]`

Reply:
expect differences within 2 knots of the break points

`cubic2` is globally coupled so may need looser tols everywhere

Reply:
This resolved itself, not sure how. But, now the same thing is happening with "cubic" and "length"

Reply:
I think it is because of this. That test passes with `grid = LinearGrid(N=100)` and not `grid = LinearGrid(N=500)`